### PR TITLE
Tilemap editor plugins

### PR DESCRIPTION
**Pull request by britzl**
This pull request adds a plugin system for the tilemap editor, where plugins can hook into the paint and clear tile functionality of the tilemap editor. Plugins will be notified when a tile is painted or cleared and the plugin can return a list of additional tiles which should be changed based on the change to the tilemap. This can be used to implement auto tiling and similar functionality.

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/defold/extension/editor/ITilemapPlugin.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.defold.extension.editor;
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/defold/extension/editor/ITilemapPlugin.java:1]: Questionable if the interface for editor plugins should be located here in bob?

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/plugin/PluginScanner.java [hunk 2]
```diff
@@ -45,13 +54,14 @@ public class PluginScanner {
 	 * This function will generate a CompileExceptionError if more than one
 	 * class in the package path extends the same base class.
 	 * 
+	 * @param scanner
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/plugin/PluginScanner.java:57]: Need to be able to pass classloader scanner from the editor

**Comment by EsdrasCaleb** [comment]
this is abandoned?

**Comment by britzl** [comment]
> this is abandoned?

Yes, unfortunately. We need to redesign the solution.

**Pull request closed without merging**